internal/restorable/images_js.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2017 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build js
+// +build js
+
+package restorable
+
+import (
+	"archive/zip"
+	"bytes"
+	"image"
+	"syscall/js"
+
+	"github.com/hajimehoshi/ebiten/v2/internal/graphicsdriver"
+)
+
+// DumpImages dumps all the current images to the specified directory.
+//
+// This is for testing usage.
+func DumpImages(graphicsDriver graphicsdriver.Graphics, dir string) error {
```

Comment:
I'm afraid I don't request `EBITENGINE_INTERNAL_IMAGES_KEY` change. If we would do we should do this in another PR.

Reply:
removing that part now

Reply:
actually, `EBITENGINE_INTERNAL_IMAGES_KEY` works without modifying restorable at all - it just prompts a download of each image - but you do need to remove the mkdir in imagedumper to avoid a syscall error